packages/discount-code-exporter/test/main.spec.js

```diff
@@ -1,11 +1,220 @@
+import DiscountCodeExport from '../src/main'
 
-describe('findSum', () => {
-  it('should be a function', () => {
-    expect(typeof findSum).toBe('function')
+describe('DiscountCodeExport', () => {
+  const logger = {
+    error: () => {},
+    warn: () => {},
+    info: () => {},
+    verbose: () => {},
+  }
+
+  let codeExport
+  beforeEach(() => {
+    codeExport = new DiscountCodeExport({
+      apiConfig: {
+        projectKey: 'asafaelhn',
+      },
+    }, logger)
+  })
+
+  describe('::constructor', () => {
+    it('should be a function', () => {
+      expect(typeof DiscountCodeExport).toBe('function')
+    })
+
+    it('should set default properties', () => {
+      expect(codeExport.apiConfig).toBeDefined()
+      expect(codeExport.batchSize).toBeDefined()
+      expect(codeExport.client).toBeDefined()
+      expect(codeExport.logger).toEqual(logger)
+      expect(codeExport.delimiter).toBe(',')
+      expect(codeExport.multiValueDelimiter).toBe(';')
+    })
+
+    it('should throw if no `apiConfig` in `options` parameter', () => {
+      expect(() => new DiscountCodeExport({ foo: 'bar' })).toThrow(
+        /The contructor must be passed an `apiConfig` object/,
+      )
+    })
+
+    it('should throw if `batchSize` is more than 500', () => {
+      expect(() => new DiscountCodeExport({
+        apiConfig: {},
+        batchSize: 501,
+      }, logger))
+      .toThrow(
+        /The `batchSize` must not be more than 500/,
+      )
+    })
+  })
+
+  describe('::run', () => {
+    it('should fetch discount codes and output csv to stream', (done) => {
+      codeExport.exportFormat = 'csv'
+      const sampleCode = {
+        code: 'discount-code',
+        name: { en: 'some-discount-name' },
+        cartDiscounts: [{ id: 'cart-discount-1' }, { id: 'cart-discount-2' }],
+      }
+      const spy = jest
+        .spyOn(codeExport, '_fetchCodes')
+        .mockImplementation((csvStream) => {
+          csvStream.write(sampleCode)
+          return Promise.resolve()
+        })
+      const outputStream = streamtest['v2'].toText((error, result) => {
+        const expectedResult = `code,name.en,cartDiscounts
+discount-code,some-discount-name,cart-discount-1;cart-discount-2`
+        expect(result).toEqual(expectedResult)
+        spy.mockRestore()
+        done()
+      })
+      codeExport.run(outputStream)
+    })
+
+    it('should fetch codes and output json to stream by default', (done) => {
+      const sampleCode = {
+        code: 'discount-code',
+        name: { en: 'some-discount-name' },
+        cartDiscounts: [{ id: 'cart-discount-1' }, { id: 'cart-discount-2' }],
+      }
+      const spy = jest
+        .spyOn(codeExport, '_fetchCodes')
+        .mockImplementation((jsonStream) => {
+          jsonStream.write(sampleCode)
+          return Promise.resolve()
+        })
+      const outputStream = streamtest['v2'].toText((error, result) => {
+        const expectedResult = [{ ...sampleCode }]
```

Comment:
why do you need to spread here?
`expectedResult = [sampleCode]` should be enough?